frontend/src/api/RestaurantApi.tsx

```diff
@@ -28,7 +34,7 @@ export const useSearchRestaurants = (city?: string) => {
   };
 
   const { data: results, isLoading } = useQuery(
-    ['searchRestaurants'],
+    ['searchRestaurants', searchState],
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/39#issuecomment-2406781721>#7f995c</a><span> - 10/11/2024, 08:04 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The React Query cache key does not include the 'city' parameter, which could lead to incorrect caching behavior when the city changes. A similar issue was also found in frontend/src/api/RestaurantApi.tsx (line 37).
</div>
<div id="fix">
<b>Fix:</b> Include the 'city' parameter in the React Query cache key to ensure that the cache is properly invalidated when the city changes.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -36,2 +36,2 @@
    const { data: results, isLoading } = useQuery(
 -    ['searchRestaurants', searchState],
 +    ['searchRestaurants', searchState, city],
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged